src/sagemaker/tensorflow/estimator.py

```diff
@@ -321,6 +323,17 @@ def _validate_args(
                     )
                 )
 
+        if (not self._script_mode_enabled()) and self._only_script_mode_supported():
+            logger.warning(
+                "Legacy mode is deprecated in versions 1.13 and higher. Using script mode instead."
+            )
+            self.script_mode = True
+
+    def _only_script_mode_supported(self):
+        return [
+            int(s) for s in self.framework_version.split(".")
+        ] >= self._LOWEST_SCRIPT_MODE_ONLY_VERSION
```

Comment:
This logic does not work for [2, 0]

Reply:
i think this logic works with [2,0]. it won't work if we have letters in the version. but we haven't run into that with tf so far.

```
>>> [2,0] > [1,13]
True
```